recipe/build.sh

```diff
@@ -1,16 +1,30 @@
 #!/usr/bin/env sh
+
+set -euxo pipefail
+
 # Get an updated config.sub and config.guess
 cp $BUILD_PREFIX/share/libtool/build-aux/config.* ./build-aux
 cp $BUILD_PREFIX/share/libtool/build-aux/config.* ./libcharset/build-aux
-set -ex
+
+# refresh the flags
+if [[ "${target_platform}" == osx-* ]]; then
+    mv lib/flags.h lib/flags.h.bak
+    ${CC_FOR_BUILD} ${CFLAGS} lib/genflags.c -o genflags
+    ./genflags > lib/flags.h
+    rm -f genflags
+    # Debugging: Show generated diff
+    diff -u lib/flags.h.bak lib/flags.h || true
+    # Check that UTF-8.MAC is included
+    grep utf8mac lib/flags.h
+fi
 
 ./configure --prefix=${PREFIX}  \
             --host=${HOST}      \
             --build=${BUILD}    \
             --enable-static     \
             --disable-rpath
```

Comment:
`Makefile.devel` has the script to re-generate `lib/flags.h` -- we may want to use that to make things simpler here.
We could only run `make -f Makefile.devel lib/flags.h` or just `make -f Makefile.devel` -- all sources should be recognized as being up to date and thus only `lib/flags.h` should be generated on macOS and nothing on Linux.
```suggestion

./configure --prefix=${PREFIX}  \
            --host=${HOST}      \
            --build=${BUILD}    \
            --enable-static     \
            --disable-rpath

# refresh the flags (we add UTF8-MAC on osx-* targets via a patch)
make -f Makefile.devel
```

Reply:
I was actually planning on carrying the new entry in the flags header as a patch in order to keep things simpler.

Reply:
Yes, that's also possible.
Letting `lib/flags.h` be generated during the build has the advantage that we have a smaller patch to maintain.
I just checked and Homebrew and MacPorts also just use `Makefile.devel`:
- https://github.com/Homebrew/homebrew-core/blob/9bd7ed750a8a2792c800f167f1d6aace5d52eabb/Formula/lib/libiconv.rb#L41
- https://github.com/MarcusCalhoun-Lopez/macports-ports/blob/5e48fc7182f11be5686c329f162e8f50288b5dee/textproc/libiconv/Portfile#L57

... and while reading that I saw that I previously didn't notice that the makefile overrides `CC`/`CFLAGS`, i.e.,
```sh
make -f Makefile.devel CC="${CC}" CFLAGS="${CFLAGS}"
```
is what we'd need actually.

(But I'm fine with either approach with only advising on trying to reduce the amount of maintenance needed.)

Reply:
Right. My concern is that the generated edit to the header won't be consistent with the other bits of the patch at some point. However, that might never happen.

Reply:
I think you are right about following the commands above to generate the changes.